applied-crypto-examples/src/merlin_non_interactive_proof.rs

```diff
@@ -6,31 +6,39 @@ use curve25519_dalek::{
-/// an integer acting as a generator of a cyclic group or a point in an elliptic curve group. The
-/// public key `K` is defined as `K = kG`.
-/// 2. Prover chooses a random scalar `a` and computes `A = aG` and sends it to the verifier.
-/// 3. Verifier defines a challenge scalar `c` and sends it to the prover
-/// 4. Prover computes the response `r` as `r = a + c*k` and sends it to the verifier
-/// 5. Verifier computes `R = rG` and `R' = A + c*K` and if `R = R'`, the proof is valid
+/// The proof demonstrated would not be suitable for production use as it is susceptible to known
+/// attacks, but it demonstrates how to define a transcript protocol and subsequently use it to
+/// perform out a non-interactive proof.
+
+/// In a proof of private key, there are 2 parties the "prover" who owns the private key `k` and the
+/// "verifier" who verifies the "prover" owns the key.
+///
+/// In the interactive case, the proof is as follows:
+/// 1. A generator point `G` is selected within the group used to perform the proof math. This is
+/// often either an integer within a cyclic group or a point in an elliptic curve group. The public
+/// key `K` is defined as `K = k*G`.
+/// 2. The Prover chooses a random scalar `a` and computes `A = a*G` and sends it to the verifier.
+/// 3. The Verifier defines a challenge scalar `c` and sends it to the prover
+/// 4. The Prover computes the response `r` as `r = a + c*k` and sends it to the verifier
+/// 5. The Verifier computes `R = r*G` and `R' = A + c*K` and if `R = R'`, the proof is valid
 ///
 /// Merlin Transcripts allow us to define a non-interactive version of this proof by allowing
 /// both parties to compute a deterministic challenge scalar `c`. To do this a transcript protocol
-/// that the verifier both agree on is defined including domain separators, an `append_proof_value()`
-/// function that serializes proof values into bytes in a canonical way and a `get_challenge()`
-/// which transforms the bytes into a scalar in a canonical way. After this is defined the proof
-/// works as follows:
-/// 1. Prover chooses a random scalar `a` and computes `A = aG` and absorbs `A` into a Merlin
+/// that the verifier both agree on is defined. To define a proof both the prover and the verifier
+/// would agree on a set of domain separators for different steps in the proof process and scheme
+/// for encoding all mathematical objects in the proof in a canonical way.
+///
+/// In the example below of a transcript protocol defined for non-interactive proofs, domain
+/// separators are created for different proof steps, and two crucial functions are defined:
+/// * `append_proof_value()`- a function that serializes proof values into bytes in a canonical
+/// * `get_challenge()` - a function that transforms the bytes into a scalar in a canonical way.
+///
+/// After this is defined the proof works as follows:
+/// 1. The Prover chooses a random scalar `a` and computes `A = aG` and absorbs `A` into a Merlin
```

Comment:
This is better!

Reply:
Yeah I attempted to make split up the example specifics from the general idea